File: scala/src/main/scala/tddmicroexercises/tirepressuremonitoringsystem/Sensor.scala
package tddmicroexercises.tirepressuremonitoringsystem

import java.util.Random

// The reading of the pressure value from the sensor is simulated in this implementation.
// Because the focus of the exercise is on the other class.

class Sensor {

     def popNextPressurePsiValue() : Double = {
        var pressureTelemetryValue : Double = 0
        pressureTelemetryValue = Sensor.samplePressure()

        return Sensor.OFFSET + pressureTelemetryValue
    }

}

object Sensor  {

  val OFFSET :  Double = 16

  private def samplePressure() : Double = {
         // placeholder implementation that simulate a real sensor in a real tire
         val basicRandomNumbersGenerator = new Random()
         val pressureTelemetryValue = 6 * basicRandomNumbersGenerator.nextDouble() * basicRandomNumbersGenerator.nextDouble()
         return pressureTelemetryValue
     }
}

File: kotlin/TirePressureMonitoringSystem/src/main/kotlin/tddmicroexercises/tirepressuremonitoringsystem/Sensor.kt
package tddmicroexercises.tirepressuremonitoringsystem

import java.util.*

// The reading of the pressure value from the sensor is simulated in this implementation.
// Because the focus of the exercise is on the other class.

class Sensor {

    fun popNextPressurePsiValue(): Double {
        val pressureTelemetryValue: Double
        pressureTelemetryValue = samplePressure()

        return OFFSET + pressureTelemetryValue
    }

    companion object {
        val OFFSET = 16.0

        private fun samplePressure(): Double {
            // placeholder implementation that simulate a real sensor in a real tire
            val basicRandomNumbersGenerator = Random()
            return 6.0 * basicRandomNumbersGenerator.nextDouble() * basicRandomNumbersGenerator.nextDouble()
        }
    }
}
